plugins/liquid/refined-types-plugin/src/main/kotlin/arrow/meta/plugins/liquid/phases/analysis/solver/ConstraintChecks.kt

```diff
@@ -908,6 +935,116 @@ private fun <A> SolverState.yesNo(conditionVars: List<Pair<A, String>>): List<Pa
   return go(conditionVars, emptyList())
 }
 
+enum class LoopPlace {
```

Comment:
Not for this PR, but when it's a good time, I plan to do a new reorganization of functions and files and packages to avoid this file and the one for gathering constraints to grow much bigger. Perhaps separating it into feature files like  `Loops,` etc... or whatever makes the most sense.